src/ripple/app/tx/impl/Transactor.h

```diff
@@ -88,6 +147,44 @@ class Transactor
         return ctx_.tx;
     }
 
+    /////////////////////////////////////////////////////
+    /*
+    These static functions are called from invoke_preclaim<Tx>
+    using name hiding to accomplish compile-time polymorphism,
+    so derived classes can override for different or extra
+    functionality. Use with care, as these are not really
+    virtual and so don't have the compiler-time protection that
+    comes with it.
+    */
+
+    static
+    TER
+    checkSeq (PreclaimContext const& ctx);
+
+    static
+    TER
+    checkFee (PreclaimContext const& ctx, std::uint64_t baseFee);
+
+    static
+    TER
+    checkSign (PreclaimContext const& ctx);
+
+    // Returns the fee in fee units, not scaled for load.
+    static
+    std::uint64_t
+    calculateBaseFee (
+        PreclaimContext const& ctx);
+
+    static
+    TER
+    preclaim(PreclaimContext const &ctx)
```

Comment:
https://github.com/ripple/rippled/pull/1305#discussion_r40846600
> Mainly as a shortcut to avoid having to copy all the individual members of the context into the result.